File: single-number-iii.java
public class Solution {
    public int[] singleNumber(int[] nums) {
        int s = 0;
        for (int i = 0; i < nums.length; ++i) {
            s ^= nums[i];
        }
        Arrays.sort(nums);
        int res1 = 0;
        for (int i = 0; i < nums.length; i += 2) {
            if (nums[i] != nums[i+1]) {
                res1 = nums[i];
                break;
            }
        }
        int[] vec = new int[2];
        vec[0] = res1;
        vec[1] = s ^ res1;
        return vec;
        
    }
}

File: src/main/cpp/single-number-iii.cpp
class Solution {
public:
    int getNum(vector<int>& nums) {
        int s = 0;
        for (int i = 0; i < nums.size(); ++i) {
            s ^= nums[i];
        }
        return s;
    }
    vector<int> singleNumber(vector<int>& nums) {
        int s = 0;
        for (int i = 0; i < nums.size(); ++i) {
            s ^= nums[i];
        }
        int tmp = s & (-s);
        vector<int> vec1;
        vector<int> vec2;
        for (int i = 0; i < nums.size(); ++i) {
            if ((nums[i] & tmp) == 0) {
                vec1.push_back(nums[i]);
            } else {
                vec2.push_back(nums[i]);
            }
        }
        vector<int> vec;
        int res1 = getNum(vec1);
        vec.push_back(res1);
        vec.push_back(s ^ res1);
        return vec;
    }
};